### ISSUE TITLE
Cannot find ways to generate workloads for given CNN models

### ISSUE THREAD

**Issue by tangyuelm** [astra-sim/astra-sim#99]
Dear Authors,

I just installed the tool and ran it following the instructions, and now it can work well. However, I found that the workloads in the astra-sim\inputs\workload\ASTRA-sim-1.0 are pre-defined files. I also read the readme and know that each subsequent line in a file like the Resnat file describes a layer with the format of layer description including layer name, reserved variable, FP compute time, etc. However, I have no idea of how to get such information with only the CNN model and hardware platform parameters. 

I tried to find some ways to generate such workload files and found that there should be a  "scripts/workload_generator/gen_astrasim_workload_input.py" file to generate such workload mentioned in https://github.com/astra-sim/astra-sim/issues/34. However, it seems that the file is not in the installation package based on the current version. I have attached the screenshot of the hierarchy of the installed files. I also cannot find this file in the github folder. Could you please let me know where to find these codes? If it requires other installation, would it be possible to provide any instructions? Thank you.

  
![image](https://github.com/astra-sim/astra-sim/assets/58452529/a47f3855-2bb0-40d5-ab57-e3fa710dec83)

**Comment by tangyuelm** [comment]
I found there is a scripts/workload_generator/gen_astrasim_workload_input.py code is in the astra-sim 1.0 branch. However, when I try to follow the instructions to compiling & running analytical backend as the network simulator, I got the following error.
![image](https://github.com/astra-sim/astra-sim/assets/58452529/53a1456a-7f5a-4407-9905-73904399e89f)
![image](https://github.com/astra-sim/astra-sim/assets/58452529/76505b57-c10a-4ae1-84b4-336f3b68a5c7)
I tried to bypass the compilation and generate the workload files using the gen_astrasim_workload_input.py directly but also met an error shown as follows.
![image](https://github.com/astra-sim/astra-sim/assets/58452529/a9ffeb46-7c3b-43c8-8543-61ab0a42a7ca)
It seems that the SCALE-Sim should be here after I build the backend. Could anyone provide any help? Thank you.

**Comment by TaekyungHeo** [comment]
The workload generator, scripts/workload_generator/gen_astrasim_workload_input.py, is not compatible with ASTRA-sim 2.0. To generate synthetic workloads, please use the et_generator in Chakra.

**Comment by tangyuelm** [comment]
Dear @TaekyungHeo,

Thank you for the explanation. However, my problem is that given a CNN network (e.g. Conv layers with input channel, output channel, kernel size, etc.) and NPU platform (e.g. hardware constraint for each NPU and the connection between each NPU), we can get the estimated latency. From et_generator, I found that the inputs are already represented as computation and communication for each node but I have no idea to generate these information is I only have the CNN network. Thus, I found the et_converter can convert the CNN workload (e.g. ResNet) into executed .et. However, after I study the et_converter, I found the workload files have already given the computation time, communication time, etc. for each Conv layer instead of the initial Conv parameters. That's why I am wondering if there is a way to know such workload information with the Conv parameters and hardware constraints. Thank you.
